dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,78 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled to true updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    section.sharing_disabled = true
+
+    # Student update happens on before save
+    section.save
```

Comment:
nit: lines 80 and 82 can be combined into `section.update(sharing_disabled: true)`. Is there a reason to keep separate them?